java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -1507,14 +1507,15 @@ open class KotlinFileExtractor(
                 }
                 is IrFunction -> {
                     if (s.isLocalFunction()) {
-                        val classId = extractGeneratedClass(s, listOf(pluginContext.irBuiltIns.anyType))
+                        val compilerGeneratedKind = if (s.origin == IrDeclarationOrigin.ADAPTER_FOR_CALLABLE_REFERENCE) {
+                            CompilerGeneratedKinds.DECLARING_CLASSES_OF_ADAPTER_FUNCTIONS
+                        } else {
+                            null
```

Comment:
It feels unintuitive for me that we pass `null`, but in the end it becomes `CompilerGeneratedKinds.CALLABLE_CLASS`. Should we instead explicitly specify the latter here?

Reply:
I considered that, but then we'd have to repeat `CALLABLE_CLASS` in 3 places. I think of the null here as "no special value; just use the default". Perhaps I should have called it `compilerGeneratedKindOverride` instead?

Reply:
I see. Another option would be specifying a default parameter value. But `compilerGeneratedKindOverride` also sounds reasonable.

Reply:
The problem with a default parameter value is I think I'd need to specify it for both overloads of the function, as well as passing it as a parameter in the call that makes use of the "last argument is a block" syntax. So I've gone with the `...Override` name to avoid repeating it.